chapters/en/slides/module7/module7_03.md

```diff
@@ -0,0 +1,389 @@
+
+## It is more useful to bind a slider to a dataframe column not displayed on the chart axes
+
+``` python
+slider = alt.binding_range(name='Year')
+select_rating = alt.selection_single(fields=['Release_Year'], bind=slider)
+
+points.encode(
+    opacity=alt.condition( select_rating, alt.value(0.7), alt.value(0.1))
+).add_selection(select_rating)
+```
+
+<iframe src="/module7/charts/03/unnamed-chunk-9.html" width="100%" height="420px" style="border:none;">
+</iframe>
+
+Notes: So far we have used the same column for the slider as we have on
+the x-axis to clearly see the effect from changing the slider behavior.
+
+However, when creating an interactive visualization it is often the most
+useful to bind the slider to a dataframe column that is not already
+included on one of the axes.
+
+In this slide we try to use the slider widget to highlight movies based
+on the year they were produced, so that we can effectively explore
+movies from different time periods separately.
+
+However, as you can see it doesn’t work as we expected and nothing
+happens when we drag the slider handle back and fourth. The reason for
+this is that sliders don’t work well with temporal data in Altair.
+
+As an alternative to the slider, we can use another plot to to select
+data, which has several other advantages as you will see in the next
+slide.
+
+---
+
+## Driving slider-like selections from another plot is more effective
+
+``` python
+select_year = alt.selection_interval()
+bar_slider = (alt.Chart(movies).mark_bar().encode(
+    alt.X('Release_Year', title='Release Year'),
+    alt.Y('count()'),
+    opacity=alt.condition(select_year, alt.value(0.7), alt.value(0.1)))
+.properties(height=50).add_selection(select_year))
+
+points.encode(opacity=alt.condition(select_year, alt.value(0.7), alt.value(0.1))) & bar_slider
+```
+
+<iframe src="/module7/charts/03/unnamed-chunk-10.html" width="100%" height="420px" style="border:none;">
+</iframe>
+
+Notes: Instead of having a a slider, it is more effective to use a
+histogram for the selection since this is also gives us information
+about the distribution of the movies over the years.
+
+We can use what we learned in the last slide deck to link the selection
+between the two plots. The biggest difference here is that instead of a
+single or multi selection, we will use the `selection_interval` function
+to create a range selection.
+
+In this histogram we can see that there are just a few movies from the
+1980s in the data and only one in 2040, which must be an error in the
+dataset.
+
+In addition to showing the distribution of the data this also allows us
+to select a range of values, which is not possible with sliders in
+Altair. We can then drag the selection with the mouse, just as when we
+were selecting in the scatter plots in the first slide deck of this
+module.
+
+If we wanted to we could also style the histogram chart to look more
+like a widget, and remove the incorrect data point, but we won’t do that
+here.
```

Comment:
```suggestion
Notes: In our temporal data case, where we want to filter by year, 
instead of having a a slider, it is more effective to use a
histogram that we can use for selecting a range of years, because
this also gives us information about how many movies we have 
data for each year.

We can use what we learned in the last slide deck to link the selection
between the two plots. The biggest difference here is that instead of a
single or multi selection, we will use the `selection_interval` function
to create a range selection.

In this histogram we can see that there are just a few movies from the
1980s in the data and only one in 2040, which must be an error in the
dataset.

In addition to showing the distribution of the data this also allows us
to select a range of values, which is not possible with sliders in
Altair. We can then drag the selection with the mouse, just as when we
were selecting in the scatter plots in the first slide deck of this
module.

If we wanted to we could also style the histogram chart to look more
like a widget, and remove the incorrect data point, but we won’t do that
here.
```